esx_service/vmdk_ops_test.py

```diff
@@ -1082,6 +1084,87 @@ def test_vmdkops_on_different_tenants(self):
+        error_info = auth_api._tenant_access_add(name=self.tenant1_name, 
+                                                 datastore=self.datastore1_name, 
+                                                 allow_create=False,
+                                                 switch_default_datastore=False, 
+                                                 volume_maxsize=volume_maxsize, 
+                                                 volume_totalsize=volume_totalsize)
+
+        # set access privileges on self.datastore, with allow_create=True
+        # keep the default_datastore to self.datastore1 for tenant1
+        error_info = auth_api._tenant_access_add(name=self.tenant1_name, 
+                                                 datastore=self.datastore_name, 
+                                                 allow_create=True,
+                                                 switch_default_datastore=False, 
+                                                 volume_maxsize=volume_maxsize, 
+                                                 volume_totalsize=volume_totalsize)
+                    
+        # create a volume with default size
+        # a volume will be tried to create on the default_datastore, which is self.datastore1_name
+        # create should fail since on default_datastore, it does not have "allow_create"" set to True
+        opts={u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol1_name, opts)
+        self.assertNotEqual(None, error_info)
+
+        # try to create this volume on self.datastore, which has "allow_create" set to True
+        # since self.datastore is not the "default_datastore" for tenant1
+        # we should use full_vol_name like vol_name@datastore_name
+        full_name_vol1 = self.tenant1_vol1_name + "@" + self.datastore_name
+        opts={u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, full_name_vol1, opts)
+        self.assertEqual(None, error_info)
+
+        # switch "default_datastore" to self.datastore
+        error_info = auth_api._tenant_update(name=self.tenant1_name, 
+                                             default_datastore=self.datastore_name) 
+        self.assertEqual(None, error_info)
+
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol2_name, opts)
+        self.assertEqual(None, error_info)
+
+        # list volumes
+        opts = {}
+        result = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, 'list', None, opts)
+        
+        # there should be two volumes shown as "tenant1_vol1"" and "tenant1_vol2"
+        self.assertEqual(len(result), 2)
+        self.assertEqual("tenant1_vol1", result[0]['Name'])
+        self.assertEqual("tenant1_vol2", result[1]['Name'])
```

Comment:
Is it checking that  @default_datastore is stripped from the name?
Then pls ad this to the comments